db/db_impl.cc

```diff
@@ -747,6 +753,10 @@ SequenceNumber DBImpl::IncAndFetchSequenceNumber() {
   return versions_->FetchAddLastToBeWrittenSequence(1ull) + 1ull;
 }
 
+void DBImpl::SetPreserveDeletesSequenceNumber(SequenceNumber seqnum) {
+  preserve_deletes_seqnum_.store(seqnum);
```

Comment:
can we assert that it's monotonically increasing?

Reply:
actually i'd rather check and return false if `preserve_deletes_seqnum_` hasn't been updated as a result of the call. Failing an assert in the response to incorrect user input seems wrong; user is responsible for checking that the call always returns true as it should if the user input is sane.

Reply:
sure, let's fail the call in that case.